src/ec/curve25519/ops.rs

```diff
@@ -121,6 +121,79 @@ fn encode_point(x: Elem, y: Elem, z: Elem) -> EncodedPoint {
     bytes
 }
 
+// If the target architecture is non-Windows x86-64 or NEON-capable ARM, it is
+// faster to do scalar base point multiplication using the generic scalar
+// multiplication implementation.
+#[cfg(all(target_arch = "x86_64", not(windows)))]
+pub fn scalar_mult_with_base_point(scalar: &Scalar) -> EncodedPoint {
```

Comment:
This is an `EncodedPoint` in the X25519 sense, not the Ed25519 sense, right? It would be nice to find a way to make this clearer.